### ISSUE TITLE
curly: multi-or-nest should allow lexical declarations to have braces for single-statement blocks

### ISSUE THREAD

**Issue by cherryblossom000** [eslint/eslint#11663]
**What rule do you want to change?**
curly

**Does this change cause the rule to produce more or fewer warnings?**
fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
The `multi-or-nest` option should allow single-statement blocks with lexical declarations (`let`, `const`, `function`, and `class`) to have braces.

**Please provide some example code that this change will affect:**

```js
if (foo) {
  const bar = 'baz';
}
```

**What does the rule currently do for this code?**
ESLint outputs the error "Unnecessary { after 'if' condition." if the `multi-or-nest` option is set. After this is fixed:
```js
if (foo)
  const bar = 'baz';
```
it outputs the error "Parsing error: Unexpected token const" as lexical declarations need to be in a block.

**What will the rule do after it's changed?**
The `multi-or-nest` option would allow the code and not output any errors.

**Are you willing to submit a pull request to implement this change?**
I would if I could, but I'm not great at coding.

**Comment by platinumazure** [comment]
Hi @cherryblossom000, thanks for the issue.

This seems like a bug-- we shouldn't report in that case if "fixing" the issue will result in a parse error.

To be clear, does ESLint's autofix feature actually apply a fix in this case?

**Comment by cherryblossom000** [comment]
Yes, sorry for not clarifying earlier. ESLint autofixed it to become
```js
if (foo)
  const bar = 'baz';
```
which produces a parsing error.

Even if the autofix problem is solved, wouldn't the `multi-or-nest` option produce an error for
```js
if (foo) {
  const bar = 'baz';
}
```
because single-statement blocks can't have braces?

**Comment by platinumazure** [comment]
> Even if the autofix problem is solved, wouldn't the multi-or-nest option produce an error[...]

I think it would and that's why I think we shouldn't report an error there at all. As a bonus, if no error is reported, then no fixing will take place.

**Comment by platinumazure** [comment]
Marking as "accepted" as I can [recreate the issue](https://eslint.org/demo/#eyJ0ZXh0IjoiLyogZXNsaW50IGN1cmx5OiBbZXJyb3IsIFwibXVsdGktb3ItbmVzdFwiXSAqL1xyXG5cclxuaWYgKGZvbykge1xyXG4gIGNvbnN0IGJhciA9ICdiYXonO1xyXG59Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo2LCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19) in our online demo.

**Comment by BRKurek** [comment]
Working on this
